kernel/src/snapshot.rs

```diff
@@ -385,6 +400,53 @@ impl Snapshot {
 
         domain_metadata_configuration(self.log_segment(), domain, engine)
     }
+
+    /// Get the In-Commit Timestamp of this snapshot.
+    ///
+    /// This method only works when In-Commit Timestamps (ICT) is enabled for the table.
+    /// It returns the inCommitTimestamp from the CommitInfo action of the commit that created this snapshot.
+    ///
+    /// Returns an error if ICT is not enabled for this table.
+    pub fn get_in_commit_timestamp(&self, engine: &dyn Engine) -> DeltaResult<i64> {
+        // Check if we have a cached value first
+        if let Some(cached_timestamp) = self.latest_in_commit_timestamp.get() {
+            return match *cached_timestamp {
+                Some(ict) => Ok(ict),
+                None => Err(Error::generic(
+                    "In-Commit Timestamps are not available for this snapshot",
+                )),
+            };
+        }
+
+        let enablement_info = self
+            .table_configuration()
+            .in_commit_timestamp_enablement()?;
+
+        let computed_timestamp = if let Some((enablement_version, _)) = enablement_info {
+            // ICT is enabled for the table
+            if self.version() < enablement_version {
+                // This snapshot predates ICT enablement
+                None
+            } else {
+                // This snapshot should have ICT, call utility on ParsedLogPath to do the rest of the work
+                let commit_path = ParsedLogPath::new_commit(self.table_root(), self.version())?;
+                commit_path.extract_in_commit_timestamp(engine)?
+            }
+        } else {
+            // ICT was never enabled for this table
+            None
+        };
+
+        // Cache the result
+        let _ = self.latest_in_commit_timestamp.set(computed_timestamp);
+
+        match computed_timestamp {
+            Some(ict) => Ok(ict),
+            None => Err(Error::generic(
+                "In-Commit Timestamps are not available for this snapshot",
+            )),
+        }
+    }
```

Comment:
Make sure that error message has enough context that can help us debug :)
```suggestion
       computed_timestamp.ok_or_else(|| {Error::generic(
                "In-Commit Timestamp was enabled at version {snapshot.version()}, but was not present.",
            ),
        });
```